### PR TITLE
init: `maestro` dev automations

### PR DESCRIPTION
**Pull request by siddarthkay**
## Summary

This PR introduces `maestro` dev automations.
The purpose of this PR is to allow running `maestro` automations to enhance productivity by **automating** mundane tasks.

Other flows like 
- send contact request
- sync devices
- join a community 
etc will be a part of follow up PRs.

This PR also exists to facilitate a discussion whether the team finds this useful or not.
Any/all feedback is welcome.

In this PR we provide a `make` command to run a very common task like creating account & login : 
`make auto-login`

We also allow another `make` command to run any other custom flow : 
`make auto-custom FLOW="maestro/create-account-or-login.yaml"`

A `maestro` folder has been added with 3 `yaml` files which are very easy to understand 
ref : https://maestro.mobile.dev/api-reference/commands

## Review notes
follow these steps and share your experience
- `make run-clojure`
- `make run-metro`
- `make run-ios` OR `make run-android`
- `make automate-login`

## Testing notes
doesn't need testing, this PR is only for devs to automate few tasks locally.

## Platforms
- Android
- iOS

## Screen recording

https://github.com/status-im/status-mobile/assets/64726664/62002796-df55-4051-8228-4a293a4fb629

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

**Review by mohsen-ghafouri — APPROVED** [review]
Brilliant 💪 🚀

**Review by FFFra — APPROVED** [review]

**Comment by FFFra** [comment]
Amazing work!

File: maestro/onboarding.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+appId: im.status.ethereum.debug
+---
+- tapOn: "I’m new to Status"
+- tapOn: "small-option-card"
+- tapOn: "Your name"
+- inputText: "DeviceA"
```
Comment (siddarthkay) [inline maestro/onboarding.yaml:6]: we could take inspiration from these examples to generate random usernames and passwords : https://github.com/mobile-dev-inc/maestro/blob/main/samples/wikipedia-android-advanced/scripts/generateCredentials.js#L2

File: maestro/create-account-or-login.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+appId: im.status.ethereum.debug
+---
+- runFlow: # to run onboarding flow only when device is in onboarding state
+      when:
+        visible: "I’m new to Status"
+      file: onboarding.yaml
+
+- runFlow: # to run login flow only when device is on login screen
+      when:
```
Comment (siddarthkay) [inline maestro/create-account-or-login.yaml:9]: this `when` clause is known to slow things down.
Maybe it makes sense to keep this here and introduce separate `make automate-login` and `make automate-onboarding` 🤔

File: maestro/onboarding.yaml (hunk 1 above)
Comment (siddarthkay) [reply maestro/onboarding.yaml:6]: I will work on this in follow ups

File: maestro/create-account-or-login.yaml (hunk 2 above)
Comment (siddarthkay) [reply maestro/create-account-or-login.yaml:9]: I will work on this in follow ups

File: Makefile [hunk 3]
```diff
@@ -457,3 +457,18 @@ repl-clojure: ##@repl Start Clojure repl for mobile App
...
+#--------------
+# Dev Automation Flows
+#--------------
+
+automate-login: export TARGET := default
+automate-login: ##@other runs flow for login or onboarding app on simulator/emulator
+	maestro test "maestro/create-account-or-login.yaml"
+
+automate-custom: export TARGET := default
+automate-custom: ##@other runs any custom maestro automation flow on simulator/emulator
```
Comment (jakubgs) [inline Makefile]: Maybe a shorter `auto-*` prefix would be easier to type.

Also are we sure this should go into `other` section of help? Maybe it should have it's own `auto` section.

**Review by jakubgs — APPROVED** [review]

**Pull request merged**